Refactor Hyperfoil document

diegolovison opened the pull request.

willr3 commented on docs/tutorial/hyperfoil.adoc:
````diff
@@ -1,129 +1,156 @@
+
+Run with and replace `targethost` value with your needs
+```
+targethost=$USER@localhost
````
why does the guide introduce the extra step of creating the environment variable? Is this necessary?

diegolovison replied:
Changed

willr3 commented on docs/tutorial/hyperfoil.adoc:
````diff
@@ -1,129 +1,156 @@
-The previous tutorial created a qDup yaml that downloaded Quarkus getting started and ran a single curl against the `quarkus dev` process. We are going to start where it left off:
-
-```yaml
+== Running the experiment
````
Please add a sentence after the heading as a code section immediately after the heading can leave a user to scroll past the code section to find out why the code section exists then scrolling back up to the code section to interact with it.

diegolovison replied:
Done

willr3 commented on docs/tutorial/hyperfoil.adoc:
```diff
@@ -1,129 +1,156 @@
 == Splitting the roles
-The previous tutorial created an `example` role to run all of ours scripts. This works but a proper performance test will normally isolate the load generation (currently `curl`) from the process we are testing (currently `quarkus dev`).
-qDup isolates scripts by assigning them to roles that are running on different `hosts`. To isolate `test-endoing` from `getting-started` we need to introduce a second role. Let us call it `hyperfoil`. The `roles` section will now look like the following:
+The previous tutorial created a role to run all of ours scripts. This works but a proper performance test will normally isolate the load generator (`hyperfoil`) from the process we are testing (`quarkus dev`).
```
(e.g. `hyperfoil`)

diegolovison replied:
Removed `(`hyperfoil`)` and `(`quarkus dev`)`

willr3 commented on docs/tutorial/hyperfoil.adoc:
```diff
@@ -1,129 +1,156 @@
-
-The first command is to `wget` the hyperfoil release but we are going to start with `cd /tmp` like we did in `getting-started` so that we don't add files to the current home directory.
+==== setup-hyperfoil
+It will download, install Hyperfoil locally and configure the `single-request.hf.yaml` file to use `http://localhost:8080` instead of `http://hyperfoil.io`
```
The first sentence after a new heading should probabably use a proper noun as the topic of the first sentence instead of a pronoun. Starting with `it` will force a reader to go back to the previous section to identify the topic of the new section.

diegolovison replied:
Done

willr3 commented on docs/tutorial/hyperfoil.adoc:
````diff
@@ -1,129 +1,156 @@
 ```
 
-The next command is to run `> bin/cli.sh`. Test that in terminal and notice how it starts an interactive shell and changes the prompt.
+==== test-endpoint
````
What is the benefit of breaking this section into multiple sub-sections with the `====` decorator? I do not understand how breaking each script into a different heading section helps the readability. Was it difficult to read in the previous format?

diegolovison replied:
> What is the benefit of breaking this section into multiple sub-sections with the ==== decorator?

It is clear that `==== test-endpoint` belongs to `=== Hyperfoil role`

> Was it difficult to read in the previous format?

It wasn't.

Here is an example:

![image](https://github.com/user-attachments/assets/f7b303f9-9e6f-4759-b5e6-60fca4137897)

willr3 reviewed (commented):
I found this revision a bit harder to understand. I will take a look at the new rendered version; perhaps looking at just the diff is why I felt so confused.

diegolovison commented:
> I will take a look at the new rendered version

For this situation, the rendered version is easy to read.

----------

I have addressed all the comments. If you agree with the proposed solution, don't forget to squash it before merging.

![image](https://github.com/user-attachments/assets/378641e4-5db0-4a2b-aba1-8c9eec52691b)

willr3 commented on docs/tutorial/hyperfoil.adoc:
````diff
@@ -1,129 +1,153 @@
-
-```yaml
+== Running the experiment
+The script will clone the `quarkus-quickstarts` repository, install `quarkus-cli` if needed, and, start the application. In parallel, it will download and install link:https://hyperfoil.io[Hyperfoil]. Once the application is ready, it will start a simple benchmark. In the end, it will upload the result and stop the `quarkus-quickstarts` application.
````
> it will upload the result 
I think you mean it will download the result to qDup

diegolovison replied:
Done
